<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1357</width>
    <height>332</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QProgressBar" name="SendingProgress">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="value">
     <number>24</number>
    </property>
   </widget>
   <widget class="QPushButton" name="Exit">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Exit</string>
    </property>
   </widget>
   <widget class="QListWidget" name="LogWindow">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>20</y>
      <width>581</width>
      <height>211</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Log</string>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>301</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Binary</string>
    </property>
    <widget class="QPushButton" name="OpenBinFile">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>OpenBinFile</string>
     </property>
    </widget>
    <widget class="QPushButton" name="ReadBinFile">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>ReadBinFile</string>
     </property>
    </widget>
    <widget class="QPushButton" name="CloseBinFile">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>CloseBinFIle</string>
     </property>
    </widget>
    <widget class="QListWidget" name="ShowBinaryFile">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>20</y>
       <width>121</width>
       <height>111</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QToolBox" name="toolBox">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>20</y>
      <width>411</width>
      <height>251</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="Serial">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>411</width>
       <height>181</height>
      </rect>
     </property>
     <attribute name="label">
      <string>Serial Flasher</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>331</width>
        <height>161</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="RefreshPorts">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>RefreshPorts</string>
       </property>
      </widget>
      <widget class="QComboBox" name="ListPorts">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>20</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="OpenPort">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>OpenPort</string>
       </property>
      </widget>
      <widget class="QPushButton" name="ClosePort">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>60</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>ClosePort</string>
       </property>
      </widget>
      <widget class="QPushButton" name="EraseMem">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>151</width>
         <height>29</height>
        </rect>
       </property>
       <property name="text">
        <string>EraseMemory</string>
       </property>
      </widget>
      <widget class="QPushButton" name="SendBinary">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>100</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>SendBinary</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>411</width>
       <height>181</height>
      </rect>
     </property>
     <attribute name="label">
      <string>UDP Flasher</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>331</width>
        <height>191</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QComboBox" name="CB_LocalIp">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="PB_Bind">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>PB_Bind</string>
       </property>
      </widget>
      <widget class="QPushButton" name="PB_Unbind">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>PB_Unbind</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="LE_IPInput">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>20</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="inputMask">
        <string>000.000.000.000</string>
       </property>
       <property name="text">
        <string>000.000.000.000</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="LE_PortInput">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>60</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>LE_PortInput</string>
       </property>
       <property name="maxLength">
        <number>32767</number>
       </property>
      </widget>
      <widget class="QPushButton" name="SendBinaryUDP">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>140</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>SendBinaryUDP</string>
       </property>
      </widget>
      <widget class="QPushButton" name="PB_EraseMemoryUDP">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>140</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>PB_EraseMemoryUDP</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="PB_SaveLog">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>240</y>
      <width>181</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>PB_SaveLog</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1357</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSTM32_Binary_Flasher">
    <property name="title">
     <string>STM32 Binary Flasher</string>
    </property>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuSTM32_Binary_Flasher"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
